autofit/non_linear/optimize/abstract_optimize.py

```diff
@@ -1,28 +1,24 @@
-import json
+from abc import ABC
 
 from autoconf import conf
-
-from autofit.non_linear.abstract_search import NonLinearSearch
 from autofit.non_linear import samples as samp
+from autofit.non_linear.abstract_search import NonLinearSearch
+
 
-class AbstractOptimizer(NonLinearSearch):
+class AbstractOptimizer(NonLinearSearch, ABC):

```

Comment:
If a class does not implement all the abstract methods of its parent(s) it should be marked as ABC